### PR TITLE
Connect database mongo

### LINKED ISSUE #41: Create mongo atlas database
Cause we need to register our user, register them in mongodb atlas

### PR DESCRIPTION
**Pull request by Nokomi6**
Closes #41 !

### CODE REVIEW

File: server/go.mod
Comment (MelvinCou) [inline server/go.mod:1]: go.mod looks weard. Run `go mod tidy`

File: server/controllers/user.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+func CreateUserHandler(ctx *gin.Context) {
+	// var user User
+
+	// err := json.NewDecoder(r.Body).Decode(&user)
+	// if err != nil {
+	// 	http.Error(w, "Bad request", http.StatusBadRequest)
+	// 	return
+	// }
+
+	err := createUser(User{
```
Comment (MelvinCou) [inline server/controllers/user.go:23]: That's ok for now.

File: server/configs/database.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	// Create a background context
+	ctx := context.Background()
+
+	// Set the server API options
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
+	opts := options.Client().ApplyURI(uri).SetServerAPIOptions(serverAPI)
+
+	// Connect to the MongoDB server
+	client, err := mongo.Connect(ctx, opts)
```
Comment (MelvinCou) [inline server/configs/database.go:51]: Returning a new client every request may use a lot of ressources.
Use a singleton https://refactoring.guru/fr/design-patterns/singleton/go/example

**Review by MelvinCou — CHANGES_REQUESTED** [review]

File: server/configs/database.go (hunk 2 above)
Comment (Nokomi6) [reply server/configs/database.go]: Check last push for see the singleton, I test it

**Review by MelvinCou — APPROVED** [review]

**Pull request merged**